packages/react-component-manifests/src/manifests/charts/GranttChart/GranttChart.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { forwardRef, useMemo } from "react";
+import reactSchemaId from "@atrilabs/react-component-manifest-schema?id";
+import type { ReactComponentManifestSchema } from "@atrilabs/react-component-manifest-schema/lib/types";
+import iconSchemaId from "@atrilabs/component-icon-manifest-schema?id";
+import { CommonIcon } from "../../CommonIcon";
+import { CustomPropsTreeOptions } from "@atrilabs/app-design-forest/lib/customPropsTree";
+import CustomTreeId from "@atrilabs/app-design-forest/lib/customPropsTree?id";
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+} from "recharts";
+import { CSSTreeOptions } from "@atrilabs/app-design-forest/lib/cssTree";
+import CSSTreeId from "@atrilabs/app-design-forest/lib/cssTree?id";
+import { ReactComponent as Icon } from "../PieChart/icon.svg";
+export const GranttChart = forwardRef<
+  HTMLDivElement,
+  {
+    styles: React.CSSProperties;
+    custom: {
+      cartesianGrid?: { show?: boolean; strokeDasharray?: string };
+      data: {
+        [key: string]: number | string | Date;
+      }[];
+      options?: {
+        margin?: {
+          top: number;
+          right: number;
+          left: number;
+          bottom: number;
+        };
+      };
+      toolTip?: { show?: boolean };
+      legend?: { show?: boolean };
+      keys?: { value?: string };
+      xAxis?: { show?: boolean; key?: string; type?: "category" | "number" };
+      yAxis?: { show?: boolean; type?: "category" | "number" };
+      chartWidth: number;
+      chartHeight: number;
+    };
+    className?: string;
+  }
+>((props, ref) => {
+  const convertTasksToRechartsData = (tasks: any) => {

```

Comment:
Why are you using `tasks: any`? Please provide a well defined type.

Reply:
Also, you should this function `convertTasksToRechartsData ` outside the React Component.